analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AdditionalPropertiesManipulationModelCache.java

```diff
@@ -0,0 +1,35 @@
+package org.jboss.gm.analyzer.alignment;
+
+import org.gradle.api.Project;
+import org.gradle.api.plugins.ExtraPropertiesExtension;
+import org.jboss.gm.common.alignment.ManipulationModel;
+import org.jboss.gm.common.alignment.ManipulationModelCache;
+
+/**
+ * Delegates to the additional properties stored in Project
+ * This ensures that all tasks will have the same view of the cache
+ */
+public class AdditionalPropertiesManipulationModelCache implements ManipulationModelCache {
+
+    private static final String NAME_PREFIX = "manipulationModelCache";
+
+    private final Project project;
+
+    public AdditionalPropertiesManipulationModelCache(Project project) {
+        this.project = project;
+    }
+
+    @Override
+    public ManipulationModel get(String name) {
+        try {
+            return (ManipulationModel) AdditionalPropertiesUtil.getProperty(project, NAME_PREFIX + name);
+        } catch (ExtraPropertiesExtension.UnknownPropertyException e) {
+            return null;
+        }
+    }
+
+    @Override
+    public void put(String name, ManipulationModel manipulationModel) {
+        AdditionalPropertiesUtil.setPropertyIfMissing(project, NAME_PREFIX + name, manipulationModel);
```

Comment:
Should we allow overriding? I can imagine being surprised my value not being stored if the property already existed.

Reply:
Indeed, but this is very internal and pretty much only used here. I don't mind adding it however